### ISSUE TITLE
Create RealWorld application

### ISSUE THREAD

**Issue by talentlessguy** [talentlessguy/tinyhttp#147]
Create a real-world application, e.g. a Medium clone, according to [this repo](https://github.com/gothinkster/realworld) spec

Stack:

| tool        | name                        |
|------------|-----------------------------|
| web framework | tinyhttp (obv) |
| body parser       | [milliparsec](https://github.com/talentlessguy/milliparsec)    |
| database | PostgreSQL |
| ORM (ik it's not but it's easier to call it like this | [Prisma](https://www.prisma.io) |
| template engine | [Eta](https://github.com/eta-dev/eta) |
| auth | JWT (w/ `@tinyhttp/jwt`) |
| cors | `@tinyhttp/cors` |

The whole code must be in a separate repositor and the PR only needs to include a link to that repository.

example readme:

## RealWorld

Example tinyhttp Medium clone following the [RealWorld](https://github.com/gothinkster/realworld) spec.

Check the [node-tinyhttp-realworld-example-app](example.com) repository.

**Comment by saranshabd** [comment]
@talentlessguy Can the frontend and backend be developed separately? If yes, then I can build a food ordering application using React.js on frontend and Node.js on the backend using the above-mentioned stack.

**Comment by talentlessguy** [comment]
@saranshabd frontend and backend should be the same app. For rendering things use template engines. React can also be one. But I'd recommend to go with Eta

**Comment by talentlessguy** [comment]
@saranshabd lol, "real world" is not something abstract, it's a name of a repo with real world apps. checks the links in the issue description. "Any" app won't fit. it must strictly follow the RealWorld spec.

**Comment by saranshabd** [comment]
@talentlessguy My bad. I did check out the real-world repo but I assumed we can build any real-world application and not just Medium. I can build Medium clone as well (it would be actually simpler than a food ordering system since I won't have to deal with transactions).

**Comment by saranshabd** [comment]
For a single frontend and backend application, we could also use [Next.js](https://nextjs.org/) or [Gatsby](https://www.gatsbyjs.com/), instead of using a template engine

**Comment by talentlessguy** [comment]
@saranshabd it won't fit the RealWorld spec then

and using such frameworks as Next.js or Gatsby or whatever else won't prove the usability of tinyhttp. App must be server side only. in the real-world repo, there are lots of examples w/ similar tasks, including Express example.

**Comment by saranshabd** [comment]
@talentlessguy Both Next.js and Gatsby render pages on the server-side. I'm not sure about Gatsby but Next.js even allows custom server instead of their default ones (which in this case will be `tinyhttp`). But that is not the point here, it was just a suggestion I made.

**Comment by talentlessguy** [comment]
@saranshabd I know that Next.js has SSR, I even used to write Next.js + Koa in the past. Stiill, the app must use only template engines, no any other web framework. Next.js is already a web framework. The focus should be made on tinyhttp.

**Comment by saranshabd** [comment]
@talentlessguy Cool, I can work with the above tech stack as well, or at least give it my best shot. So, do you want to assign this task to me?

_PS: This is the first time I am trying to contribute to open source 😅_

**Comment by talentlessguy** [comment]
@saranshabd I can try to assign this task to you, but I think it will only make sense if you do the task properly, e.g. follow the spec and use the specified technologies.

I hope I don't sound harsh, I just want people to work productively w/o wasting time

**Comment by saranshabd** [comment]
@talentlessguy Thanks! Do you have a Discord channel or a Slack workspace where I can talk to you later regarding some issues or if I want to ask something? Or does that all happen on GitHub?

**Comment by thebearingedge** [comment]
@talentlessguy Hi, I'd like to give this a shot. Should this be written in JavaScript or is TypeScript ok?

**Comment by talentlessguy** [comment]
@thebearingedge I think it doesn't matter that much here but I prefer JS so you can show that ESM works with Node and tinyhttp without any custom build step

**Comment by thebearingedge** [comment]
Ok, I'm on it. I noticed that you included the `eta` template engine, but the specs from RealWorld are for a REST API. It's cool if we just submit a JSON back end?

**Comment by talentlessguy** [comment]
@thebearingedge oops my bad

yeah eta is not needed here

**Comment by talentlessguy** [comment]
@thebearingedge ping

how is it going with the realworld app?

(not hurrying ya or anything, jsut checking if everything is alright and you're not stuck bc of tinyhttp bugs)

**Comment by thebearingedge** [comment]
Hey! Yeah, it's going ok. I'm not going to lie, the initial ramp up on a 100% new set of tech was pretty rough in the beginning 😅 . I haven't used an ORM in about 5 years.

But I think I have some time today to get a couple more routes done.

[Here is where the project](https://github.com/thebearingedge/node-tinyhttp-realworld-example-app) is right now. The code will need to get rearranged quite a bit because some of the choices make no sense in the context of this setup, particularly since Prisma does not support Long Running Transactions, which is how I normally handle tests 🤷‍♀️ 

Anyway, it's been pretty fun so far! 👍

**Comment by talentlessguy** [comment]
@thebearingedge nice! great to see that everything is going well

**Comment by ac5tin** [comment]
> @talentlessguy Hi, I'd like to give this a shot. Should this be written in JavaScript or is TypeScript ok?

Hi, can you also produce a Typescript version please?

**Comment by talentlessguy** [comment]
@thebearingedge hi, I have a little question about the status of the repo

I'm planning to release V2 in the near future (1-3 months) and have a question

will the real-world application be ready by that time? it's completely fine if it won't, don't worry about that. I'm just asking in case I could include that in the release notes of V2. otherwise, it'll be left for next release

**Comment by thebearingedge** [comment]
@talentlessguy Yeah, I have some PTO coming up in a few weeks 👍

**Comment by aldy505** [comment]
@thebearingedge hello, do you need help?

**Comment by thebearingedge** [comment]
@aldy505 that would be fantastic! 👍 

I probably won't have time to dedicate to the project until about two weeks.

**Comment by aldy505** [comment]
@thebearingedge I'll open an issue on your repo about which part will I be working.

**Comment by jeanhdev** [comment]
Hi guys, is the project on still ? I'd like to contribute

**Comment by talentlessguy** [comment]
@jeanhdev yes it is and if you want you could fork that real world example or start your own from scratch, using the stack mentioned in readme

**Comment by rezendevictor** [comment]
@talentlessguy can I give a shot at this? Any specifications not present in this thread that I need to know before I start?

**Comment by talentlessguy** [comment]
> @talentlessguy can I give a shot at this? Any specifications not present in this thread that I need to know before I start? 

yeah sure, no specifications besides the RealWorld official guide + just use some of tech stack I suggested + create your own repo and I'll later transfer it to tinyhttp repo with you as an owner
